components/public-api-server/pkg/apiv1/tokens_test.go

```diff
@@ -364,16 +366,66 @@ func TestTokensService_RegeneratePersonalAccessToken(t *testing.T) {
 		require.Equal(t, connect.CodeInvalidArgument, connect.CodeOf(err))
 	})
 
-	t.Run("unimplemented when feature flag enabled", func(t *testing.T) {
-		serverMock, _, client := setupTokensService(t, withTokenFeatureEnabled)
+	t.Run("responds with not found when token is not found", func(t *testing.T) {
+		serverMock, dbConn, client := setupTokensService(t, withTokenFeatureEnabled)
+
+		someTokenId := uuid.New().String()
+
+		dbtest.CreatePersonalAccessTokenRecords(t, dbConn,
+			dbtest.NewPersonalAccessToken(t, db.PersonalAccessToken{
+				UserID: uuid.MustParse(user.ID),
+			}),
+			dbtest.NewPersonalAccessToken(t, db.PersonalAccessToken{
+				UserID: uuid.MustParse(user2.ID),
+			}),
+		)
 
 		serverMock.EXPECT().GetLoggedInUser(gomock.Any()).Return(user, nil)
 
-		_, err := client.RegeneratePersonalAccessToken(context.Background(), connect.NewRequest(&v1.RegeneratePersonalAccessTokenRequest{
-			Id: uuid.New().String(),
+		newTime, err := time.Parse(time.RFC3339, "2023-01-02T15:04:05Z")
+		require.NoError(t, err)
+
+		newTimestamp := timestamppb.New(newTime)
+		_, err = client.RegeneratePersonalAccessToken(context.Background(), connect.NewRequest(&v1.RegeneratePersonalAccessTokenRequest{
+			Id:             someTokenId,
+			ExpirationTime: newTimestamp,
```

Comment:
```suggestion
			ExpirationTime: timestampb.New(time.Date(2023, 1, 2, 15, 4, 5, 0, 0, time.UTC))
```

If you wanted a terser version, you can do this. It has the benefit of not having to parse (and check for err)